pymatgen/core/structure.py

```diff
@@ -2669,6 +2682,10 @@ def from_file(cls, filename, primitive=False, sort=False, merge_tol=0.0):
             from pymatgen.io.fleur import FleurInput
 
             s = FleurInput.from_file(filename).structure
+        elif fnmatch(fname, "*.res"):
```

Comment:
@ScottNotFound Thanks for this PR! Very nice work!

I just tried `Structure.from_file` on a local `.res` file and get an unexpected error

```py
import os

from pymatgen.core import Structure


res_file = "NaLiTa2O6-28290-4142-1.res"
print(os.path.isfile(res_file))
>>> True
Structure.from_file(res_file)
>>> FileNotFoundError: [Errno 2] No such file or directory: 'NaLiTa2O6-28290-4142-1.res'
```

Maybe worth adding a test case for that to see if it also fails?

Reply:
<details>
<summary>full stack trace</summary>

```py
File ~/dev/pmg/pymatgen/core/structure.py:2688, in IStructure.from_file(cls, filename, primitive, sort, merge_tol)
   [2685](file:///Users/janosh/dev/pmg/pymatgen/core/structure.py?line=2684) elif fnmatch(fname, "*.res"):
   [2686](file:///Users/janosh/dev/pmg/pymatgen/core/structure.py?line=2685)     from pymatgen.io.res import ResIO
-> [2688](file:///Users/janosh/dev/pmg/pymatgen/core/structure.py?line=2687)     s = ResIO.structure_from_file(fname)
   [2689](file:///Users/janosh/dev/pmg/pymatgen/core/structure.py?line=2688) else:
   [2690](file:///Users/janosh/dev/pmg/pymatgen/core/structure.py?line=2689)     raise ValueError("Unrecognized file extension!")

File ~/dev/pmg/pymatgen/io/res.py:618, in ResIO.structure_from_file(cls, filename)
    [613](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=612) @classmethod
    [614](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=613) def structure_from_file(cls, filename: str) -> Structure:
    [615](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=614)     """
    [616](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=615)     Produces a pymatgen Structure from a res file.
    [617](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=616)     """
--> [618](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=617)     return ResProvider.from_file(filename).structure

File ~/dev/pmg/pymatgen/io/res.py:344, in ResProvider.from_file(cls, filename)
    [341](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=340) @classmethod
    [342](file:///Users/janosh/dev/pmg/pymatgen/io/res.py?line=341) def from_file(cls, filename: str):
...
     [43](file:///Users/janosh/.venv/py310/lib/python3.10/site-packages/monty/io.py?line=42) if ext in (".XZ", ".LZMA"):
     [44](file:///Users/janosh/.venv/py310/lib/python3.10/site-packages/monty/io.py?line=43)     return lzma.open(filename, *args, **kwargs)
---> [45](file:///Users/janosh/.venv/py310/lib/python3.10/site-packages/monty/io.py?line=44) return open(filename, *args, **kwargs)

FileNotFoundError: [Errno 2] No such file or directory: 'NaLiTa2O6-28290-4142-1.res'
```
</details>

Reply:
Strange error, I think a bunch of that stack trace got truncated. I assume it's pointing to [here](https://github.com/materialsvirtuallab/monty/blob/50adcaf72844b44aab088dc502298cec4f3fa51c/monty/io.py#L45). Not sure how this would fail unless the file actually wasn't present?

Reply:
Oops. The file did in fact not exist because I used fname rather than filename to read it.

Reply:
Nice, working now! 👍